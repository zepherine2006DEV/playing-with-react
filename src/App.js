import React from 'react';
import './App.css';

function App() {
  function sayHi() {
    alert("Hello!!!")
    //return <p>Hi Computer</p>
  }
  
  return (
    <div>
      <h1>Hello World</h1>
      <button onClick={sayHi}>Hi</button>
    </div>
  )
}
export default App;
