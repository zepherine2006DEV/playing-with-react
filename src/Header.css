.header{
  background : green;
  color : yellow;
}