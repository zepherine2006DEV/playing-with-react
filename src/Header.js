import React from 'react';
import './Header.css';

function Header(){
  return (
    <div  className='header'>
      <h1>Hello World</h1>
    </div>
  );
}

export default Header;
