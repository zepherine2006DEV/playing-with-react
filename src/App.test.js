import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Header from './Header';
import renderer from 'react-test-renderer';
import { testNameToKey } from 'jest-snapshot/build/utils';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// it('renders "Hello World!', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   expect(div.textContent).toBe('Hello World');
// });

describe("Header",() =>{
  it('Displays "Hello World', () => {
    const div =document.createElement('div');
    ReactDOM.render(<Header />, div);
    expect(div.textContent).toBe('Hello World');
  });
});

test('exploring snapshot', () => {
  const component = renderer.create(<Header />);
  let result = component.toJSON();
  console.log(result);
  expect(result).toMatchSnapshot();
});