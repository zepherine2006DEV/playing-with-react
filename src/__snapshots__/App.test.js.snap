// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exploring snapshot 1`] = `
<h1>
  Hello World
</h1>
`;
